### PR TITLE
Add people to meilisearch media, and therefore searchable

### PR DESCRIPTION
**Pull request by Primdahl26**
# Description

This adds people to be searchable. This PR comes with some decisions that i am unsure if we want to continue with. For an example it adds people to the `media` index of meilisearch, but i am unsure if we want a separate one. I liked the idea of having one "master" index though.

Furthermore this adds quite a lot of API alls to `tmdb`, and makes the running through `json.gz` files take quite a lot longer since the new person file is quite long.

This also removes the `first-time` parameter of our `full-setup` since it was never used anyway. It is not triggered from providers being updated, and thus useless for us.

image

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

### CODE REVIEW

**Comment by AndreasPB** [comment]
:heart:

File: internal/types.go [hunk 1]
```diff
@@ -67,6 +67,35 @@ func (movie *Movie) toMedia() *Media {
...
+	Gender             int     `json:"gender"`
+	ImdbId             string  `json:"imdb_id"`
+	ProfilePath        string  `json:"profile_path"`
+	Popularity         float32 `json:"popularity"`
+	Birthday           string  `json:"birthday"`
+	KnownForDepartment string  `json:"known_for_department"`
+	PlaceOfBirth       string  `json:"place_of_birth"`
+	Biography          string  `json:"biography"`
+	// TODO: Add rest of person items when we migrate to edgeDB
+}
```
[the remark this answers is not in the document]
Comment (Primdahl26) [reply internal/types.go:82]: Its a part of the data under `movie_credits` or `tv_credits` under a specific movie or tv show. We would have to do something like `translation`, so we would iterate through the `tv_credits` and `movie_credits`, and create a long string, but i dont know if i like that solution 😢

File: backend/cli.py [hunk 2]
```diff
@@ -29,6 +28,8 @@
 from meilisearch.errors import MeiliSearchApiError
 from tqdm import tqdm
 
+# from app.api import fetch_changed_media_ids
+
```
Comment (AndreasPB) [inline backend/cli.py:32]: 🐴

File: frontend/src/lib/components/filters.svelte [hunk 3]
```diff
@@ -160,8 +176,9 @@
             class="btn btn-xs {filterAmount ? 'btn-primary' : 'btn-disabled'}"
             on:click={() => {
               $filters.minImdb = 0
-              $filters.tvChecked = true
-              $filters.movieChecked = true
+              for (const key in $filters.checked) {
+                $filters.checked[key] = false
+              }
```
Comment (AndreasPB) [inline frontend/src/lib/components/filters.svelte:181]: We might have talked about it, but I feel like this should work

```suggestion
              for (let key of $filters.checked) {
                key = false
              }
```

File: frontend/src/routes/+page.svelte [hunk 4]
```diff
@@ -45,27 +45,40 @@
...
-    // Example: "?g=Action&g=Comedy&g=Drama"
-
     if (!currentMediaAmount) {
       setViewportToDefault()
     }
 
+    // Builds the optional query for genres
+    // Example: "?g=Action&g=Comedy&g=Drama"
     let query = ""
+    let userInput = input.replace(/[^\w\d\s\&]/g, " ")
```
Comment (AndreasPB) [inline frontend/src/routes/+page.svelte:55]: We should probably add a comment with something like:
```suggestion
    // Sanitizes input by removing any character that is not a letter, number, whitespace, or an ampersand
    // and replaces them with a space
    let userInput = input.replace(/[^\w\d\s\&]/g, " ")
```

File: internal/routes.go [hunk 5]
```diff
@@ -66,13 +67,20 @@ func processIds(c *gin.Context) {
 				fetchTV(id[1:], TVCh)
 				<-guardCh
 			}(id, tvCh)
+		case "p":
+			go func(id string, personCh chan Person) {
+				defer wg.Done()
+				fetchPerson(id[1:], personCh)
+				<-guardCh
+			}(id, personCh)
```
Comment (AndreasPB) [inline internal/routes.go:75]: We should probably talk about how we split this monster of a function up

**Review by AndreasPB — COMMENTED** [review]
I love the extra refactoring that happened - lovely little facelift :D 

I'll wait with the approve till we've heard from some of our users, and have discussed some of the things from my review

File: frontend/src/lib/components/filters.svelte (hunk 3 above)
Comment (AndreasPB) [reply frontend/src/lib/components/filters.svelte:181]: Didn't work 🤷

**Review by AndreasPB — APPROVED** [review]

**Pull request merged**